### DISCUSSION TITLE
Use public DNS servers for dynamic filtering

### DISCUSSION THREAD

**Discussion by DimitrisKalaboukas** [uBlockOrigin/uBlock-issues#3437] [Ideas] [answered]
I wonder if it is a good idea to combine public DNS services with powerful ad, tracking, and malware filtering capabilities, like AdGuard DNS, CleanBrowsing Security Filter, ControlD, and/or dns0.eu (ZERO), in order to give uBO the ability to dynamically improve its blocking performance by accessing in real-time the most up-to-date domain blocklists. In a new tab named “DNS filters”, the user could select which such services uBO would consult in parallel during browsing. If a selected DNS server doesn’t resolve a domain request, then it will be blocked.

**Comment by MasterKia** [comment]
https://github.com/uBlockOrigin/uBlock-issues/wiki/About-%22Why-uBlock-Origin-works-so-much-better-than-Pi%E2%80%91hole-does%3F%22

**Reply by garry-ut99** [reply]
Yes, by the way also here some other guy asked similiar question and got same answer: https://github.com/uBlockOrigin/uBlock-issues/issues/3309#issuecomment-2308747116.

**Comment by Yuki2718** [comment, marked as answer]
uBO does not connect to any server except for filter lists: https://github.com/gorhill/ublock/wiki/Privacy-policy Who takes responsibility if one of these providers turned to be malicious (including but not limited to hijacking)?

**Reply by DimitrisKalaboukas** [reply]
There are really some privacy concerns...

**Reply by gorhill** [reply]
Just set your DNS resolver to one of these DNS server you trust, this doesn't have to be done through uBO, and this will work for all your devices.

**Comment by DimitrisKalaboukas** [comment]
[Osprey: Browser Protection](https://github.com/Foulest/Osprey) is an open-source extension that does exactly this.
